core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -436,30 +456,66 @@ export class StateGraphLayoutService {
         fractionalGridHeight * offsetInGridRectangles);
     };
 
-    for (var nodeId in nodeData) {
-      nodeData[nodeId].y0 = getVerticalPosition(
-        nodeData[nodeId].depth + GRID_NODE_Y_PADDING_FRACTION);
-      nodeData[nodeId].x0 = getHorizontalPosition(
-        nodeData[nodeId].offset + GRID_NODE_X_PADDING_FRACTION);
+    for (var nodeId in nodeDataTemp) {
+      nodeDataTemp[nodeId].y0 = getVerticalPosition(
+        nodeDataTemp[nodeId].depth + GRID_NODE_Y_PADDING_FRACTION);
+      nodeDataTemp[nodeId].x0 = getHorizontalPosition(
+        nodeDataTemp[nodeId].offset + GRID_NODE_X_PADDING_FRACTION);
 
-      nodeData[nodeId].yLabel = getVerticalPosition(
-        nodeData[nodeId].depth + 0.5);
-      nodeData[nodeId].xLabel = getHorizontalPosition(
-        nodeData[nodeId].offset + 0.5) + X_LABEL_OFFSET_CHECKPOINT_ICON;
+      nodeDataTemp[nodeId].yLabel = getVerticalPosition(
+        nodeDataTemp[nodeId].depth + 0.5);
+      nodeDataTemp[nodeId].xLabel = getHorizontalPosition(
+        nodeDataTemp[nodeId].offset + 0.5) +
+        X_LABEL_OFFSET_CHECKPOINT_ICON;
 
-      nodeData[nodeId].height = (
+      nodeDataTemp[nodeId].height = (
         (1.0 - VERTICAL_EDGE_PADDING_FRACTION * 2) / totalRows
       ) * (1.0 - GRID_NODE_Y_PADDING_FRACTION * 2);
-      nodeData[nodeId].width = (
+      nodeDataTemp[nodeId].width = (
         (1.0 - HORIZONTAL_EDGE_PADDING_FRACTION * 2) / totalColumns
       ) * (1.0 - GRID_NODE_X_PADDING_FRACTION * 2) +
       WIDTH_OFFSET_CHECKPOINT_ICON;
     }
 
-    // Assign id and label to each node.
-    for (var nodeId in nodeData) {
-      nodeData[nodeId].id = nodeId;
-      nodeData[nodeId].label = nodes[nodeId];
```

Comment:
@vojtechjelinek I have assigned these `id` and `label` at the time of initialization itself. 
**Refer lines: 267-268**